BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -239,54 +244,60 @@ open class SBAProfileItemBase: NSObject, SBAProfileItem {
             return self.commonDemographicJsonValue()
         }
     }
+    
+    open var readonly: Bool {
+        return false
+    }
+}
+
+extension SBAKeychainWrapper: SBAKeychainWrapperProtocol {
 }
 
 open class SBAKeychainProfileItem: SBAProfileItemBase {
-    private let keychain: SBAKeychainWrapper
+    
+    var keychain: SBAKeychainWrapperProtocol
     
     public required init(dictionaryRepresentation dictionary: [AnyHashable: Any]) {
-        var keychainToUse = SBAUser.shared.keychain
         
-        let keychainService = dictionary["keychainService"] as! String?
-        let keychainAccessGroup = dictionary["keychainAccessGroup"] as! String?
+        // Look to see if this item uses a different keychain service or access group
+        // than the default and instantiate if it does.
+        let keychainService = dictionary["keychainService"] as? String
+        let keychainAccessGroup = dictionary["keychainAccessGroup"] as? String
         
         if (keychainService != nil || keychainAccessGroup != nil) {
-            keychainToUse = SBAKeychainWrapper(service: keychainService, accessGroup: keychainAccessGroup)
+            keychain = SBAKeychainWrapper(service: keychainService, accessGroup: keychainAccessGroup)
+        }
+        else {
+            // Otherwise, use the default
+            keychain = SBAProfileManager.keychain
         }
-        
-        keychain = keychainToUse
         
         super.init(dictionaryRepresentation: dictionary)
     }
     
     override open var value: Any? {
         get {
-            var err: NSError?
-            let obj = keychain.object(forKey: profileKey, error: &err)
-            if let error = err {
-                print("Error accessing keychain \(profileKey): \(error.code) \(error)")
-            }
-            return self.typedValue(from: obj)
+            return storedValue(forKey: sourceKey)
         }
         
         set {
             do {
                 if newValue == nil {
-                    try keychain.removeObject(forKey: profileKey)
+                    try keychain.removeObject(forKey: sourceKey)
                 } else {
                     if !self.commonCheckTypeCompatible(newValue: newValue) {
-                        assert(false, "Error setting \(profileKey): \(String(describing: newValue)) not compatible with specified type \(itemType.rawValue)")
+                        assert(false, "Error setting \(sourceKey): \(String(describing: newValue)) not compatible with specified type \(itemType.rawValue)")
```

Comment:
I'm thinking this particular message maybe should refer to both the profileKey and the sourceKey to avoid confusion about what it means.